src/extension.ts

```diff
@@ -176,6 +180,19 @@ function initializeIconPaths(context: vscode.ExtensionContext) {
     ext.iconPaths.light.schema = context.asAbsolutePath('resources/light/schema.svg')
 }
 
+function updateMruWhenAwsContextChanges(awsContext: AwsContext, extensionContext: vscode.ExtensionContext) {
+    extensionContext.subscriptions.push(
+        awsContext.onDidChangeContext(async awsContextChangedEvent => {
+            if (!awsContextChangedEvent.profileName) {
+                return
+            }
+
+            const mru = new CredentialsProfileMru(extensionContext)
```

Comment:
This feels weird creating a new CredentialProfileMru every time you update the credentials. Should this either be a const created further up the chain or a functional method (not really seeing much reason why this needs to be class-based)?

Reply:
I was on the fence about tearing down the class, because it seems like some functions would be sufficient. Because that MRU class is backed by the extension context's global state, it makes sense to be global (or singleton instance) style. 

Open to making that change here or punting. Seems like it would be better handled as a separate change.